Source/WebKit/WPEPlatform/wpe/WPEToplevel.cpp

```diff
@@ -126,6 +175,29 @@ void wpeToplevelRemoveView(WPEToplevel* toplevel, WPEView* view)
     toplevel->priv->views.remove(view);
 }
 
+GRefPtr<WPEView> wpeToplevelGetView(WPEToplevel* toplevel, size_t index)
+{
+    if (index >= toplevel->priv->views.size())
+        return nullptr;
+
+    auto views = copyToVectorOf<GRefPtr<WPEView>>(toplevel->priv->views);
+    return views[index];
```

Comment:
Creating a temporary vector to then extract a single items seems a bit wasteful. Why not returning the desired element directly?

Something like this:

```suggestion
    size_t current = 0;
    for (auto* view : toplevel->priv->views)
        if (current++ == index)
            return GRefPtr(view);
    ASSERT_NOT_REACHED();
```